### DISCUSSION TITLE
Polymorphic API (base and sub type models) in downstream services doesn't seem generate schemas in swagger.json

### DISCUSSION THREAD

**Discussion by assadf** [Burgyn/MMLib.SwaggerForOcelot#135] [Q&A] [answered]
Hi,

In one of my downstream services, I have an API Endpoint which has a polymorphic model (base and sub types).  When I configure the services.AddSwaggerGen in my downstream service to UseOneOfForPolymorphism(), the schemas aren't generated in the OpenAPI JSON schema hence are not displayed in the Gateway Swagger UI.  When I navigate to the Swagger UI directly to my downstream service it correctly displays the polymorphic model in the UI (and obviously the OpenAPI JSON has the various subtype models listed under schemas).  Does the SwaggerForOcelot not support UseOneOfForPolymorphism() in the downstream services?

I'm using the following versions:

- .NET Core 3.1
- Swashbuckle.AspNetCore 5.6.3
- MMLib.SwaggerForOcelot 3.0.0
- Ocelot 16.0.1

**Comment by Burgyn** [comment]
Hi,

thank you for your question. Unfortunately, this package does not support `UseOneOfForPolymorphism`.
Can you provide an example of your downstream service swagger.json? It can help me figure out how difficult it is to implement.

Thanks.

**Comment by assadf** [comment]
Thanks Burgyn for the quick reply.  

I've created a quick simple API which demonstrates the Polymorphic API endpoint (called api/events) and have added the swagger.Json in the readme.  You can access the solution at https://github.com/assadf/PolymorphicAPI.

**Comment by Burgyn** [comment, marked as answer]
Hi @assadf,

thanks for your demo. Please can you try `version 3.2.0`?

**Comment by assadf** [comment]
@Burgyn, it's working beautifully.  Thanks for the quick turnaround on this, you're a star.

**Reply by Burgyn** [reply]
You are welcome.
